src/smallmap_gui.cpp

```diff
@@ -1004,6 +1005,43 @@ class SmallMapWindow : public Window {
 		}
 	}
 
+	/**
+	 * Adds industry names to the smallmap.
+	 * @param dpi the part of the smallmap to be drawn into
+	 */
+	void DrawIndustryNames(const DrawPixelInfo *dpi) const
+	{
+		if (this->map_type != SMT_INDUSTRY) return;
+
+		for (const Industry *i : Industry::Iterate()) {
+			const LegendAndColour &tbl = _legend_from_industries[_industry_to_list_pos[i->type]];
+			if (!tbl.show_on_map) continue;
+
+			const bool is_blinking = i->type == _smallmap_industry_highlight && !_smallmap_industry_highlight_state;
+			if (is_blinking) continue;
+
+			const uint8_t legend_text_width = GetStringBoundingBox(tbl.legend, FS_SMALL).width;
+			const int vertical_padding = ScaleGUITrad(4);
+
+			/* Remap the industry coordinate */
+			const TileIndex &tile = i->location.GetCenterTile();
+			const Point pt = this->RemapTile(TileX(tile), TileY(tile));
+			const int x = pt.x - this->subscroll - (legend_text_width / 2);
+			const int y = pt.y + vertical_padding;
+
+			/* Check if the industry name is within bounds */
+			if (x + legend_text_width > dpi->left &&
+					x < dpi->left + dpi->width &&
+					y + GetCharacterHeight(FS_SMALL) > dpi->top &&
+					y < dpi->top + dpi->height) {
+
+				/* And draw it. */
+				SetDParam(0, i->index);
+				DrawString(x, x + legend_text_width, y, tbl.legend, TC_WHITE, SA_LEFT, false, FS_SMALL);
```

Comment:
Not sure if I'm using `DrawString(…)` correctly here.

The same call in `DrawTowns(…)` doesn't pass `FS_SMALL` or `TC_WHITE`; but if I omit those params here, texts are drawn with normal size in blue